conductr_cli/sandbox_features.py

```diff
@@ -27,6 +28,73 @@ def __init__(self, name, port):
         self.port = port
 
 
+class ProxyingFeature:
+    """Proxying feature.
+
+    On start, the proxying features will be run.
+    """
+
+    name = 'proxying'
+    ports = []
+    dependencies = []
+
+    def __init__(self, version_args, image_version, offline_mode):
+        self.version_args = version_args
+        self.image_version = image_version
+        self.offline_mode = offline_mode
+        self.host = None
+        self.proxy_bind_addr = None
+        self.bundle_http_port = None
+        self.proxy_ports = None
+
+    def configure(self, args, run_result):
+        self.host = run_result.host
+        self.proxy_bind_addr = run_result.core_addrs[0]
+        self.bundle_http_port = args.bundle_http_port
+        self.proxy_ports = sorted(args.ports)
+
+    @staticmethod
+    def conductr_args():
+        return []
+
+    def conductr_feature_envs(self):
+        return []
+
+    def conductr_roles(self):
+        if major_version(self.image_version) == 1:
+            return []
+        else:
+            return ['haproxy']
+
+    def start(self):
+        if major_version(self.image_version) == 1 or None in [self.host, self.proxy_bind_addr, self.bundle_http_port, self.proxy_ports]:
+            return []
+        else:
+            log = logging.getLogger(__name__)
+            log.info(h1('Starting proxying feature'))
+
+            started = sandbox_proxy.start_proxy(proxy_bind_addr=self.proxy_bind_addr,
+                                                bundle_http_port=self.bundle_http_port,
+                                                proxy_ports=self.proxy_ports)
+
+            if started:
+                log.info('HAProxy has been started')
+                log.info('Your Bundles are by default accessible on:')
+                log.info('  {}:{}'.format(self.host, CONDUCTR_BUNDLE_PROXY_PORT))
```

Comment:
I think `CONDUCTR_BUNDLE_PROXY_PORT` should be replaced in lieu of `self.bundle_http_port` as it's possible for the user to set bundle port apart from default `9000` and this value is kept in `self.bundle_http_port`.